### ISSUE TITLE
Unable to connect to Deluge server

### ISSUE THREAD

**Issue by YoonAddicting** [tympanix/Electorrent#146]
**Client:**
- [ ] µTorrent
- [ ] qBittorrent
- [ ] Transmission
- [ ] rTorrent
- [x] Deluge

**Client Version:** 1.3.15

**Operating System:** Host: FreeBSD (FreeNAS 11.2), Client: Windows 10 x64

**Application Version:** 2.6

**Description:** I am unable to setup the Deluge connection, neither with the builtin localclient user, nor with a brand new user, which is working through the Web-UI and a thin-client on the same windows client as Electorrent.
I've tried both with the local IP, remote IP and web address, both Web-UI port and deluged daemon port, https and http (https is on and forced on the server, with a valid Let's Encrypt certificate), and no combination of any of these seems to be working.
Currently the Web-UI is accessible through https://<localIP>:10276 and the daemon is running on 58846.
I'm kinda lost at where I should be looking exactly for what the different parameters should be, if they aren't the correct ones.

The exact error I am receiving is:
_Connection problem_
_The connection could not be established_

(I think Deluge also needs to be added to the default issue formula.)

**Comment by tympanix** [comment]
If you are able to open the developer console you might get more info on the error. *The connection could not be established* is just a generic error for when the type of error is not recognised. Deluge is a bit different in the way that the web UI can serve multiple daemons. I suspect you only have one daemon running? If you have any other information on how I can recreate your setup please let me know. I have personally tested with https://github.com/linuxserver/docker-deluge. I will also add the Deluge option to the issue template.

**Comment by YoonAddicting** [comment]
Attached here is an image of what the developer console says.
![image](https://user-images.githubusercontent.com/9822733/51875537-b3fecc80-2365-11e9-8e26-3e9043aa68dc.png)

I'm running a jail with Deluge on FreeNAS, running as the user _nobody_ (UID: 65534).
When connecting to the Web-UI on port 10276 for the first time it asks for a password for the Web-UI, no username however. After then I can select which daemon I'd like to connect to, where I'd connect to the one running on 127.0.0.1:58846. This daemon is running with the localclient username inside Deluge's _auth_ file, no password is required to select this one though, after the initial password to connect to the Web-UI. Inside the web.conf file on the server, it shows the session login as "_admin_".
As mentioned before, the daemon is enforcing SSL on Deluge. This is done through a Let's Encrypt certificate issued for the domain it's available through.

UPDATE Mid-ways through writing this:
I tried disabling SSL in deluge, so it would connect over HTTP, and boom it worked, Filling in the IP, protocol, password (Web-UI password), client program and port (Web-UI port). So no username for Deluge. Now the next question would be, why doesn't it want to connect over SSL.

**Comment by tympanix** [comment]
Great debugging work. So we are down to this being an SSL problem. I must admit I did not test Deluge with SSL just yet. Just to make sure, even though it says *Invalid password* in the console, I expect you to use the exact same settings (e.i. same password etc.), hence this error message is probably bogus?

**Comment by YoonAddicting** [comment]
Yup, the password is the same (and correct), just did a sanity check to be sure, so the error message should be unrelated.
The files it refers to differ between selecting HTTP and HTTPS, but both simply return Invalid password. I've attached a picture of the console in both cases, HTTPS on the top.
![image](https://user-images.githubusercontent.com/9822733/51877462-92edaa00-236c-11e9-85ef-22c49321a2e1.png)

**Comment by tympanix** [comment]
The error is thrown from here i both cases: https://github.com/tympanix/node-deluge/blob/b60695055f5991e06369d04ad1b2f2a798760ea0/index.js#L67

Some of the files are only different because HTTPS uses a different implementation underneath. From the implementation I can see that the *Invalid password* error is thrown prematurely, so something completely different may be wrong. You can set a breakpoint at the line which i linked in the developer console to check it out in detail if you wish. If not, I will get around to it myself at some point 👍

**Comment by KatherineKane** [comment]
Sorry to jump on such an old thread but this one seemed the correct path based on what I'm dealing with. I was running into this same error, but I'd already changed my protocol based on another issue thread about the problems with using https. When that change didn't solve my problem I kept hunting and eventually found this thread, which mentioned the use of the WebUI Port. My seedbox is at USB and they provide remote client instructions with a separate port number from their WebUI. 

Once I switched to the WebUI Port it connected instantly. But as soon as my torrent list loaded, it started losing the connection and then reconnecting every few seconds. Finally I had to just shut it down. But when I restart it and try again, I get the same response. I have attached a screenshot, but for the record, I am not a programmer or a coder. So this is all Greek to me. If there is some other info or troubleshooting you need me to do, explicit instructions would be appreciated.

![Snap5000](https://github.com/user-attachments/assets/dcc991ba-67e1-4996-a058-1ca308e7f3fa)

**Comment by tympanix** [comment]
HTTP timeout issues.

Fixed in #553
